go-bindata-assetfs/main.go

```diff
@@ -54,16 +51,10 @@ func assetFS() *assetfs.AssetFS {
 	}
 	panic("unreachable")
 }`)
+	// Close files BEFORE remove calls (don't use defer).
```

Comment:
In case of early failure, after opening the first FD. In our case the
process would end, closing its FDs, so in practice it doesn't matter, but
still I think defer is a good style.

Is using defer actually causing problems?

On Sat, Feb 21, 2015 at 8:46 PM, Dmitri Shuralyov <notifications@github.com>
wrote:

> In go-bindata-assetfs/main.go
> <https://github.com/elazarl/go-bindata-assetfs/pull/7#discussion_r25124458>
> :
>
> > @@ -54,16 +51,10 @@ func assetFS() *assetfs.AssetFS {
> >  	}
> >  	panic("unreachable")
> >  }`)
> > +	// Close files BEFORE remove calls (don't use defer).
>
> Why would this leak fds?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elazarl/go-bindata-assetfs/pull/7/files#r25124458>.
>